### DISCUSSION TITLE
should I update WSL 1 to WSL 2?

### DISCUSSION THREAD

**Discussion by smahdi-dev** [Cyfrin/foundry-full-course-cu#1523] [Q&A] [answered]
I have WSL already installed on my system, but it's WSL 1.
I was wandering should I use WSL 2 or it won't make any problem?

**Comment by EngrPips** [comment, marked as answer]
Hello @smahdi-dev , I would advice you stick to whatever `wsl` version that was set up in the course but if you understand how `wsl` work well enough then maybe you can use any `wsl` version of your choice as you would know whatever is going on whenever you run into issues and know how to solve them.

**Reply by smahdi-dev** [reply]
Hi @EngrPips, thanks for your advice.
I am not a `wsl` master. so I think I'll update it to `wsl 2`.

**Reply by EngrPips** [reply]
Oh okay, not bad not bad whatever works for you friend

**Comment by Niferu** [comment]
Hello @smahdi-dev 

I'm not `wsl` master either, but was facing a lot of issues and warnings while using `wsl 1` programming on `Solana` blockchain with `Anchor` framework and similar for `Foundry` (warnings mostly here), after upgrading to `wsl 2` everything got fixed, so I would recommend upgrading to `wsl 2`.

**Reply by smahdi-dev** [reply]
thanks @Nlferu bro for sharing your knowledge.

**Comment by EngrPips** [comment]
Thank you so much for the heads up @Nlferu
